### ISSUE TITLE
Add API to disable `resume=` injection

### ISSUE THREAD

**Issue by cgwalters** [rhinstaller/anaconda#1667]
See https://pagure.io/atomic-wg/issue/513

https://github.com/poncovka/anaconda/commit/07c8591b107c34f3825dfa2d2b20c5c95cb3b892 broke ostree-based systems (Atomic Host/Fedora CoreOS/Silverblue) and I don't really want to debug it and make it work, since hibernation doesn't make sense on servers, and it's already under-tested enough and marginal that I don't care about it for Silverblue either on desktops.

Maybe the simplest thing is for us to add `noresume` to the default kernel arguments, and change the code to not add `resume` if `noresume` is present. Or we could conditionalize it on the payload type being `dnf`.

**Comment by dustymabe** [comment]
There is a related discussion on [fedora devel list](https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/thread/A7S5DIUOKQPONAVXPLF5KT4EDTGQ4JH7/). Kevin had a [few more conditions](https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/message/HO4DONHUK6ILHZUSUGMF2W67WIENEIXQ/) where the `resume=` option is not useful.

**Comment by poncovka** [comment]
I think that the default resume option should be enabled or disabled in the install class and we should enable it only for Fedora Workstation. Does it sound ok?

**Comment by cgwalters** [comment]
> and we should enable it only for Fedora Workstation

That almost works except Silverblue uses that install class right now.  We could create a new one...I've been meaning to do so but it's more forking.

So I'd say that's fine, but let's also not do if the payload type is `rpmostreepayload`?

**Comment by dustymabe** [comment]
> That almost works except Silverblue uses that install class right now.

It probably should inherit from workstation but currently the silverblue installclass does not inherit from workstation:

https://github.com/rhinstaller/anaconda/blob/de9043d81196a4d244f56d1f7075c06efbd2c675/pyanaconda/installclasses/fedora_silverblue.py#L18-L25

is it valid for people to hibernate a silverblue system? i.e. does it work? 

Note that this [resume= leading to device timeouts](https://pagure.io/atomic-wg/issue/513) problem is [not currently present on silverblue](https://pagure.io/atomic-wg/issue/513#comment-536990) from my testing. So the bug isn't hitting us there, but not sure if we should or should not add resume= on those systems.

**Comment by poncovka** [comment]
Based on the code, it should be easy to add the check for the payload type, so we can do that for Fedora 29. However, Anaconda adds the resume option since Fedora 28, so I wonder what causes the current problems with booting.

**Comment by cgwalters** [comment]
> However, Anaconda adds the resume option since Fedora 28, so I wonder what causes the current problems with booting.

https://dustymabe.com/2018/10/20/rpm-ostree-bisecting-helps-track-down-boot-timeout-issue/

**Comment by dustymabe** [comment]
this is fixed in dracut for the particular issue I was seeing though I do think there are some other valid claims for making this not a default behavior. There are two options that I see:

1. make this an option that can be disabled
1. add some more heuristics for removing it when it doesn't make sense (see [devel list thread](https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/thread/A7S5DIUOKQPONAVXPLF5KT4EDTGQ4JH7/))

We could do both. In the very least I think it should be able to be disabled as the default via an installclass or an anaconda option.

**Comment by jkonecny12** [comment]
Ad 1, I don't want to add something like that to users. Most of the users would be confused why is it optional. Even I don't know when it is good idea to enable this and when not.
Ad 2, Is there a reason to have this as kernel boot param and as a part of the install class? I think one of these should be enough and I would personally prefer Install class.

What do you think @dustymabe ?

**Comment by dustymabe** [comment]
I like options. Right now this kernel arg gets added unconditionally whether a user wants it or not on x86_64 systems with swap space.

If we leave it as enabled by default:

- As a user I'd like an option (kickstart and possibly kernel `inst.` option) to disable this.
- As a user I'd also like anaconda to not apply this option when it doesn't make sense. It already only applies it on x86_64 hardware when swap exists. Kevin [spoke](https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/message/HO4DONHUK6ILHZUSUGMF2W67WIENEIXQ/) of a few more instances where it doesn't make sense. 
- As a media creator I'd like to be able to override this in the installclass (i.e. the only type of system I think this makes sense on at all is the workstation class and not any of the others).

Another alternative is that we disable it by default and only enable this for workstation.


> Most of the users would be confused why is it optional

then we can rely on the user to choose when to disable it?

**Comment by poncovka** [comment]
Just a note, so far we don't have any other reported bugs related to this issue. I would focus on removal of the option from ostree-based systems.

**Comment by sgallagher** [comment]
`resume=` is useless on Fedora Server as well, since

1. Most deployments will use Secure Boot
2. Many deployments don't have SWAP enabled (or at least not larger than the available RAM)
3. Why would anyone hibernate a server? The whole point is for them to stay running all the time.

**Comment by jkonecny12** [comment]
@sgallagher yes and that is a reason why we wanted to add this as IntallClass option so you can disable this for `SilverBlue` `Atomic` and `Server`.

**Comment by dustymabe** [comment]
> that is a reason why we wanted to add this as IntallClass option

+1 for an installclass option. Could I suggest that we default to no `resume=` and enable it only in the `workstation` installclass?

I also think we don't have to disable this on `silverblue`. I don't think this affects silverblue, but I have a machine I can check on in a bit and report back.

**Comment by jkonecny12** [comment]
I don't see any problem with having it disabled by default.

**Comment by poncovka** [comment]
Thank you for reporting this issue! The issue tracker will be closed on 3 June 2019. Please make sure that your issue is also reported at https://bugzilla.redhat.com/. We are handling a high number of issues and many of them involve other components. Bugzilla allows us to better process your issues and collaborate on them with other teams. We are sorry for the inconvenience.
